Possible panic in index code

```
=== RUN   TestSegmentDistributionDoesNotAffectQuery
! Any distribution of test documents in segments does not affect query results: Error on property evaluation after 94 passed tests: Check paniced: error
   parsing regexp: unexpected ): `.*\\)no.*` goroutine 35 [running]:
runtime/debug.Stack(0xc42c5256c8, 0x8aa920, 0xc4430db5c0)
	/home/runner/.gimme/versions/go1.10.4.linux.amd64/src/runtime/debug/stack.go:24 +0xa7
github.com/m3db/m3/vendor/github.com/leanovate/gopter.SaveProp.func1.1(0xc42c525cd0)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/prop.go:19 +0x6e
panic(0x8aa920, 0xc4430db5c0)
	/home/runner/.gimme/versions/go1.10.4.linux.amd64/src/runtime/panic.go:502 +0x229
github.com/m3db/m3/src/m3ninx/search/proptest.genRegexpQuery.func1(0xc431ee1ac0, 0x2)
	/home/runner/workspace/src/github.com/m3db/m3/src/m3ninx/search/proptest/query_gen.go:98 +0xba1
github.com/m3db/m3/vendor/github.com/leanovate/gopter/gen.OneGenOf.func1(0xc431ee1ac0, 0x10)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/gen/one_of.go:27 +0x66
github.com/m3db/m3/vendor/github.com/leanovate/gopter.Gen.Map.func1(0xc431ee1ac0, 0x5)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/gen.go:135 +0x7e
github.com/m3db/m3/vendor/github.com/leanovate/gopter/gen.OneGenOf.func1(0xc431ee1ac0, 0x8f9960)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/gen/one_of.go:27 +0x66
github.com/m3db/m3/vendor/github.com/leanovate/gopter/prop.ForAll.func1(0xc431ee1ac0, 0x961ef8)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/prop/forall.go:31 +0x140
github.com/m3db/m3/vendor/github.com/leanovate/gopter.SaveProp.func1(0xc431ee1ac0, 0x0)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/prop.go:24 +0x6c
github.com/m3db/m3/vendor/github.com/leanovate/gopter.Prop.Check.func1(0x0, 0xc430094fb0, 0xc31fe0)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/prop.go:52 +0x15c
github.com/m3db/m3/vendor/github.com/leanovate/gopter.(*runner).runWorkers(0xc431d3c090, 0x0)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/runner.go:49 +0x2d9
github.com/m3db/m3/vendor/github.com/leanovate/gopter.Prop.Check(0xc430094f90, 0xc433d806c0, 0x95a561)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/prop.go:110 +0x1af
github.com/m3db/m3/vendor/github.com/leanovate/gopter.(*Properties).Run(0xc4367aa8a0, 0x99d220, 0xc431d3a1e0, 0x4c)
	/home/runner/workspace/src/github.com/m3db/m3/vendor/github.com/leanovate/gopter/properties.go:37 +0xc9
github.com/m3db/m3/src/m3ninx/search/proptest.TestSegmentDistributionDoesNotAffectQuery(0xc42c53e2d0)
	/home/runner/workspace/src/github.com/m3db/m3/src/m3ninx/search/proptest/prop_test.go:91 +0x42d
testing.tRunner(0xc42c53e2d0, 0x9612c8)
	/home/runner/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/home/runner/.gimme/versions/go1.10.4.linux.amd64/src/testing/testing.go:824 +0x2e0


Elapsed time: 2.998481023s
--- FAIL: TestSegmentDistributionDoesNotAffectQuery (3.10s)
	prop_test.go:92: failed with initial seed: 1538058247796509198
```

--- a/src/m3ninx/search/proptest/query_gen.go
+++ b/src/m3ninx/search/proptest/query_gen.go
@@ -91,7 +91,7 @@ func genRegexpQuery(docs []doc.Document) gopter.Gen {
 
 		// escape any '(' or ')' we see to avoid regular expression parsing failure
 		escapeFront := bytes.Replace(re, []byte("("), []byte("\\("), -1)
-		escapeBack := bytes.Replace(escapeFront, []byte("("), []byte("\\)"), -1)
+		escapeBack := bytes.Replace(escapeFront, []byte(")"), []byte("\\)"), -1)
 
 		q, err := query.NewRegexpQuery(field.Name, escapeBack)
 		if err != nil {

